### DISCUSSION TITLE
Adding a disable git flag in Create Next App

### DISCUSSION THREAD

**Discussion by Arinji2** [vercel/next.js#67748] [Ideas]
### Goals

1. Adding an optional flag resembling something --disable-git in the create next app configuration

### Background

We recently had a question in the nextjs official discord regarding disabling git on setup since they didnt use git. This felt as a valid feature which is missing from the current create-next-app package. This makes nextjs much more inclusive of people/companies which don't use git. 

### Proposal

I went through the code of the create-next-app package and i am pretty sure that this part is what does the actual git setup. 
https://github.com/vercel/next.js/blob/canary/packages/create-next-app/create-app.ts#L238

I plan on adding a flag to disable git, update that code specifically and also update the docs regarding create-next-app. If this proposal does go through i would love some tips on how to actually contribute to packages, the contributions docs dosent seem to mention running packages locally.

**Comment by joulev** [comment]
Better to prompt a yes/no question asking if the user wants to initialise Git, imo. And not send that prompt if the user [is already in a Git repository](https://github.com/vercel/next.js/blob/96092dd42dc8d7fe4ef0b6ea9889020b2da2af86/packages/create-next-app/helpers/git.ts#L34).

**Reply by Arinji2** [reply]
Thing is, 99% of the times, people do want git. So it would lead to kinda an extra step. This is only for like specific cases.

**Reply by ShayokhShorfuddin** [reply]
I agree with @Arinji2 on that. A simple flag will do the job.

**Comment by ShayokhShorfuddin** [comment]
It is quite a valid suggestion. A flag will be useful when a developer doesn't intend to push code to Github.

Possible flags -
   --disable-git 
   --no-git

**Comment by Arinji2** [comment]
I am going to start on the PR and hopefully one of the team members can guide me on any errors/developing the package :D

**Comment by Arinji2** [comment]
https://github.com/vercel/next.js/pull/68821
